src/popup/pages/NetworkForm.vue

```diff
@@ -46,112 +46,140 @@
   </div>
 </template>
 
-<script>
-import { mapActions, mapGetters } from 'vuex';
-import { ROUTE_NETWORK_SETTINGS } from '../router/routeNames';
+<script lang="ts">
+import {
+  computed, defineComponent, onMounted, ref,
+} from '@vue/composition-api';
+import { ROUTE_NETWORK_EDIT, ROUTE_NETWORK_SETTINGS } from '../router/routeNames';
 import { defaultNetwork } from '../utils';
+import { useDispatch, useGetter } from '../../composables/vuex';
+import type { INetworkBase } from '../../types';
+
 import BtnMain from '../components/buttons/BtnMain.vue';
 import InputField from '../components/InputField.vue';
 import PlusCircleIcon from '../../icons/plus-circle.svg?vue-component';
 
+interface INetworkForm extends INetworkBase {
+  index?: number
```

Comment:
This property is weird in my opinion :) On my branch I'm removing it totally. Here I suggest just to add this to the INetworkBase as optional maybe.